Example

JiasuWang opened the pull request:
@dongylaq

JiasuWang committed:
Compute grad only using real arithmetic

Add opts.useReal (defaulted value is true). In QSP_solver.m, when opts.useReal = true, gradient of objective function will be computed by using only real arithmetic.

JiasuWang committed:
CM solver and Newton solver

Uploaded the code for CM solver and Newton solver.

JiasuWang committed:
Add opts.targetPre

Add an option, 'opts.target = true' means that Pre will be the target function, otherwise, Pim will equal the target function

JiasuWang committed:
Update QSPGetPimDeri_sym.m

This is the version that uses complex arithmetic and run fast. We can compare it with the real version.

JiasuWang committed:
Add CM and Newton method to QSP_solver.m

When call QSP_solver.m, opts.method determines which method will be used to obtain phase factors, opts.typePhi determines the type of output phi. Also add opts.print and opts.itprint to QSP_CM.m and QSP_Newton.m, which controls printing out the detail of solving process.

JiasuWang committed:
Add function evaluating Pre/Pim given phase factors

QSPGetEntry is intended for the check part, after obtaining phase factors using QSP_solver.m. The input opts must contain 'targetPre', 'parity' and 'typePhi'.

JiasuWang committed:
Add 2 examples

Hamiltonian simulation & matrix inversion

dongylaq committed:
change

dongylaq commented:
Use equally spaced points for numerical test rather than random points. Enough points are suggested for the test, see changes.

JiasuWang committed:
Upload 6 examples

Commit:
modify hamiltonian_simulation.m example

Commit:
modify two examples

Commit:
Move old examples to Old/ to be removed later

Commit:
move the new examples to the Examples/ folder

Commit:
rename a directory

JiasuWang committed:
Revise the code in optimization folder

Add the function of ploting the function and the approximate polynomial to cvx_poly_coef.m. plot_phase_factor_decay.m shows decay property. Revise some annotation. Rewrite QSP_CM.m and QSO_Newton.m to save one iteration.

JiasuWang committed:
Add author part

Commit:
modify Hamiltonian simulation example

Commit:
modify matrix inversion

JiasuWang committed:
Update cvx_poly_coef by addressing the edge points problem.

Commit:
edit matrix inversion and negative power

Commit:
gaussian filter

Commit:
singular vector and fix bugs in cvx_poly_coef

Commit:
singular vector

Commit:
change the target function of uniform_singular_value_amplification and
fix a few comments.

Commit:
Add an example for Chebyshev polynomial

Commit:
add a 3rd order Chebyshev polynomial example

Commit:
add a few more chebyshev examples for demonstration purpose

JiasuWang committed:
Add two examples about the quantum state preparation

Kaiser window function and Gaussian function
